hooks/usePaginatedEvents.ts

```diff
@@ -28,25 +28,24 @@ export function usePaginatedEvents(
       })
 
       if (!('error' in result)) {
-        $setHasPrevious(result.metadata.has_previous)
-        $setHasNext(result.metadata.has_next)
-        $setEvents(result.data)
+        setEvents(result)
       }
     },
-    [userId, limit]
+    [setEvents, userId, limit]
   )
 
   const fetchNext = useCallback(
-    async () => fetchEvents($events[$events.length - 1].id, 'after'),
-    [fetchEvents, $events]
+    async () =>
+      events && fetchEvents(events.data[events.data.length - 1].id, 'after'),
+    [fetchEvents, events]
   )
 
   const fetchPrevious = useCallback(
-    async () => fetchEvents($events[0].id, 'before'),
-    [fetchEvents, $events]
+    async () => events && fetchEvents(events.data[0].id, 'before'),
```

Comment:
Any need for more safety from potentially undefined values here? (not certain from reading)

Reply:
It seems like there should potentially be some bounds checking here for `undefined` when pulling values out of the arrays, but since it didn't have it before either, I'm okay shipping this for now.